frontend/routes/teamspaces/teamspaces.component.tsx

```diff
@@ -74,10 +76,12 @@ interface IProps {
 	onRevisionsClick: () => void;
 	onDownloadClick: () => void;
 	onUploadClick: () => void;
+	setState: (componentState: any) => void;
```

Comment:
You should provide `IState` as `componentState` type

Reply:
I agree with @simplypixi  if your intent is to save the whole state of the component. If you only want to save the activeTeamspace and activeProject I'd change the name of the function to be more explicit of the intent.